nullaway/src/test/java/com/uber/nullaway/VarargsTests.java

```diff
@@ -515,4 +548,33 @@ public void testVarargsRestrictive() {
             "}")
         .doTest();
   }
+
+  /**
+   * Test for a restrictive @NonNull declaration annotation on a varargs parameter defined in
+   * bytecode
+   */
+  @Test
+  public void testVarargsRestrictiveBytecodes() {
+    makeTestHelperWithArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:UnannotatedSubPackages=com.uber.lib.unannotated",
+                "-XepOpt:NullAway:AcknowledgeRestrictiveAnnotations=true"))
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import com.uber.lib.unannotated.RestrictivelyAnnotatedVarargs;",
+            "public class Test {",
+            "  public void testDeclaration() {",
+            "    String x = null;",
+            "    String[] y = null;",
+            "    // BUG: Diagnostic contains: passing @Nullable parameter 'x'",
+            "    RestrictivelyAnnotatedVarargs.test(x);",
+            "    RestrictivelyAnnotatedVarargs.test(y);",
```

Comment:
Is there a way to annotate varargs as non-null array of non-null elements? Do we want a test case for that? (separate from `RestrictivelyAnnotatedVarargs.test`)

I'd even argue that it is exceedingly rare for the intention of a varargs argument to be "a potentially nullable array of non-null elements", but I guess we should follow JSpecify here either way.

Reply:
I added this test in f30299a7facc5a04b454c132226b49800436cae4 and it exposed a bug which I fixed!  We still don't report an error when passing a `@Nullable` array for this case; that requires a full and proper fix for #1027